source/scripting/Ped.cs

```diff
@@ -1096,6 +1116,36 @@ public void ApplyDamage(int damageAmount)
 		{
 			Function.Call(Hash.APPLY_DAMAGE_TO_PED, Handle, damageAmount, true);
 		}
+		public new bool HasBeenDamagedBy(WeaponHash weapon)
```

Comment:
Instead of using `new`, change the ones in `Entity` to `virtual` and make these overrides. That's a much better solution than this.